src/lib/components/ui/button.svelte

```diff
@@ -112,59 +118,152 @@
+		}
+
+		&:hover:not(:disabled),
+		&.hover:not(:disabled) {
+			border: 1px solid var(--colors-top);
+			background: var(--colors-top);
+			color: var(--colors-base);
+		}
+
 		&:active:not(:disabled),
 		&.active:not(:disabled) {
 			border: 1px solid var(--colors-high);
 			background: var(--colors-high);
 			color: var(--colors-base);
+			outline: none;
 		}
 	}
 	.secondary {
 		border: 1px solid var(--colors-ultra-high);
 		background: none;
 		color: var(--colors-ultra-high);
 
+		&:focus:not(:disabled),
+		&:focus-visible:not(:disabled),
+		&.focus:not(:disabled) {
+			outline: 4px solid var(--colors-top);
+			outline-offset: -4px;
+			border: 1px solid transparent;
+			background: var(--colors-base);
+			color: var(--colors-top);
+		}
+
+		&:hover:not(:disabled),
+		&.hover:not(:disabled) {
+			border: 1px solid var(--colors-top);
+			background: var(--colors-low);
+			color: var(--colors-top);
+		}
+
 		&:active:not(:disabled),
 		&.active:not(:disabled) {
 			border: 1px solid var(--colors-high);
+			background: var(--colors-low);
 			color: var(--colors-high);
+			outline: none;
 		}
 	}
 	.ghost {
 		border: 1px solid transparent;
 		background: transparent;
 		color: var(--colors-ultra-high);
 
+		&:focus:not(:disabled),
+		&:focus-visible:not(:disabled),
+		&.focus:not(:disabled) {
+			outline: 4px solid var(--colors-top);
+			outline-offset: -4px;
+			background: var(--colors-base);
+			color: var(--colors-top);
+		}
+
+		&:hover:not(:disabled),
+		&.hover:not(:disabled) {
+			border: 1px solid var(--colors-low);
+			background: var(--colors-low);
+			color: var(--colors-top);
+		}
+
 		&:active:not(:disabled),
 		&.active:not(:disabled) {
 			border: 1px solid var(--colors-low);
 			background: var(--colors-low);
 			color: var(--colors-high);
+			outline: none;
 		}
 	}
 	.overlay {
 		border: 1px solid var(--colors-base);
 		background: var(--colors-base);
 		color: var(--colors-ultra-high);
 
+		&:focus:not(:disabled),
+		&:focus-visible:not(:disabled),
+		&.focus:not(:disabled) {
+			outline: 4px solid var(--colors-top);
+			outline-offset: -4px;
+			border: 1px solid transparent;
+			background: var(--colors-base);
+			color: var(--colors-top);
+		}
+
+		&:hover:not(:disabled),
+		&.hover:not(:disabled) {
+			border: 1px solid var(--colors-low);
+			background: var(--colors-low);
+			color: var(--colors-top);
+		}
+
 		&:active:not(:disabled),
 		&.active:not(:disabled) {
 			border: 1px solid var(--colors-low);
 			background: var(--colors-low);
 			color: var(--colors-high);
+			outline: none;
 		}
 	}
 	.darkoverlay {
 		border: 1px solid var(--colors-dark-overlay);
 		background: var(--colors-dark-overlay);
 		color: var(--colors-dark-top);
+
+		&:focus:not(:disabled),
+		&:focus-visible:not(:disabled),
+		&.focus:not(:disabled) {
+			outline: 4px solid var(--colors-top);
+			outline-offset: -4px;
+			border: 1px solid transparent;
+			background: var(--colors-base);
+			color: var(--colors-dark-top);
```

Comment:
In light mode the `--colors-base` and the `--colors-dark-top` is the same color, therefore the text or icon on the button is not visible.
I suggest either to use `color: var(--colors-top)` instead or use `outline: 4px solid var(--colors-dark-top)` and not change the `color` and `background`, but that also depends on the design.

Reply:
I went with the `outline: 4px solid var(--colors-dark-top)` but we will review this with @david-gauquelin